localstack/services/events/provider_v2.py

```diff
@@ -58,18 +282,329 @@ def put_rule(
+        return store
+
+    def get_event_bus(self, name: EventBusName, store: EventsStore) -> EventBus:
+        if event_bus := store.event_buses.get(name):
+            return event_bus
+        raise ResourceNotFoundException(f"Event bus {name} does not exist.")
+
+    def get_rule(self, name: RuleName, event_bus: EventBus) -> Rule:
+        if rule := event_bus.rules.get(name):
+            return rule
+        raise ResourceNotFoundException(f"Rule {name} does not exist on EventBus {event_bus.name}.")
+
+    def get_target(self, target_id: TargetId, rule: Rule) -> Target:
+        if target := rule.targets.get(target_id):
+            return target
+        raise ResourceNotFoundException(f"Target {target_id} does not exist on Rule {rule.name}.")
+
+    def get_rule_worker(
+        self, context: RequestContext, rule_name: RuleName, event_bus_name: EventBusName
+    ) -> RuleWorker:
+        store = self.get_store(context)
+        event_bus_name = self._extract_event_bus_name(event_bus_name)
+        event_bus = self.get_event_bus(event_bus_name, store)
+        rule = self.get_rule(rule_name, event_bus)
+        return self._rule_workers[rule.arn]
+
+    def create_event_bus_worker(
+        self,
+        name: EventBusName,
+        region: str,
+        account_id: str,
+        event_source_name: Optional[EventSourceName],
+        tags: Optional[TagList],
+    ) -> EventBusWorker:
+        event_bus_worker = EventBusWorker(
+            name,
+            region,
+            account_id,
+            event_source_name,
+            tags,
+        )
+        self._event_bus_workers[event_bus_worker.arn] = event_bus_worker
+        return event_bus_worker
+
+    def create_rule_worker(
+        self,
+        name: RuleName,
+        region: str,
+        account_id: str,
+        schedule_expression: Optional[ScheduleExpression],
+        event_pattern: Optional[EventPattern],
+        state: Optional[RuleState],
+        description: Optional[RuleDescription],
+        role_arn: Optional[RoleArn],
+        tags: Optional[TagList],
+        event_bus_name: Optional[EventBusName],
+        targets: Optional[TargetDict],
+    ) -> RuleWorker:
+        rule_worker = RuleWorker(
+            name,
+            region,
+            account_id,
+            schedule_expression,
+            event_pattern,
+            state,
+            description,
+            role_arn,
+            tags,
+            event_bus_name,
+            targets,
+        )
+        self._rule_workers[rule_worker.arn] = rule_worker
+        return rule_worker
+
+    def create_target_worker(
+        self, target: Target, region: str, account_id: str, rule_arn: Arn
+    ) -> TargetWorker:
+        target_worker = TargetWorkerFactory(
+            target, region, account_id, rule_arn
+        ).get_target_worker()
+        self._target_workers[target_worker.arn] = target_worker
+        return target_worker
+
+    @staticmethod
```

Comment:
moved to module level function in: https://github.com/localstack/localstack/pull/10613/commits/a5e16a99042ea595ec1d48ecd067f4c675e3a7a1

I want to keep functions that are only called within this module e.g. provider in the same file for now, since I don't want to mix v1 and v2 providers. 
At a later stage, when v1 has been moved to a separate folder it would make sense to move the helper functions into a utils file, if they are also used in a separate module. If they are only used in this module, I would still prefer to keep them in the module they are used.